services/core/MasterDriverAgent/master_driver/agent.py

```diff
@@ -400,12 +427,166 @@ def heart_beat(self):
+        revert_greenlets = []
+        stagger_interval = 0.05 #sec
+        pattern = pattern.lower()
+
+        #Add to patterns set
+        self._override_patterns.add(pattern)
+        device_topic_actual = self.instances.keys()
+        i = 0
+
+        for name in device_topic_actual:
+            name = name.lower()
+            i += 1
+            if fnmatch.fnmatch(name, pattern):
+                #If revert to default state is needed
+                if failsafe_revert:
+                    if staggered_revert:
+                        revert_greenlets.append(gevent.spawn_later(i*stagger_interval, self.instances[name].revert_all()))
+                    else:
+                        revert_greenlets.append(gevent.spawn(self.instances[name].revert_all()))
+                # Set override
+                self._override_devices.add(name)
+        config_update = False
+        # Set timer for interval of override condition
+        config_update = self._update_override_interval(duration, pattern, indefinite_duration)
+        if config_update == True:
+            #Update config store
+            patterns = dict()
+            for pat in self._override_patterns:
+                if self._override_interval_events[pat] is None:
+                    patterns[pat] = 'Indefinite'
+                else:
+                    evt, end_time = self._override_interval_events[pat]
+                    patterns[pat] = end_time.isoformat()
+            self.vip.config.set("override_patterns", jsonapi.dumps(patterns))
+
+    # Turn off override condition on all the entities matching the pattern.
+    @RPC.export
+    def set_override_off(self, pattern):
+        return self._set_override_off(pattern)
+
+    # Get a list of all the devices with override condition.
+    @RPC.export
+    def get_override_devices(self):
+        return list(self._override_devices)
+
+    # Clear all overrides
+    @RPC.export
+    def clear_overrides(self):
+        # Cancel all pending override timer events
+        for pattern, evt in self._override_interval_events.items():
+            if evt is not None:
+                evt[0].cancel()
+        self._override_interval_events.clear()
+        self._override_devices.clear()
+        self._override_patterns.clear()
+        self.vip.config.set("override_patterns", {})
+
+    # Get a list of all the devices with override condition.
+    @RPC.export
+    def get_override_patterns(self):
+        return list(self._override_patterns)
+
+    def _set_override_off(self, pattern):
+        pattern = pattern.lower()
+        config_update = False
+
+        # If pattern exactly matches
+        if pattern in self._override_patterns:
+            self._override_patterns.discard(pattern)
+            # Cancel any pending override events
+            self._cancel_override_events(pattern)
+            config_update = True
+        else:
+            _log.debug("Pattern did not match!")
```

Comment:
If this happens I think it would make more sense to raise an exception to inform the caller that something went wrong. Nothing after this needs to happen if a matching pattern was not found.